### PR TITLE
Submission for review

### PR DESCRIPTION
**Pull request by emauldin84**
All criticism is welcome!

### CODE REVIEW

File: scripts/index.js [hunk 1]
```diff
@@ -1 +1,42 @@
...
+// }
+
+// function findsACharacters(book) {
+//     aCharacters = book.filter(beginsWithZ)
+//     return aCharacters.length
+// }
+// console.log(findsACharacters(characters))
+
+// //How many characters are dead?
+deadCharacters = characters.forEach(function(key) {
```
Comment (antdevelopment1) [inline scripts/index.js]: Hey Eric were you able to get this function to output the expected number? I wasn't able to on my end. I tried to use most of the code you started with. This is a solution I tried to implement 

`function deadCount(characters) {
    let count = 0;
    characters.forEach(function(key) {
        if (key.died !== '') {
            count += 1;
        }
    })
    return count;
}

console.log(deadCount(characters))`

Let me know what you think. If yours was working on your end great job. I may have had a different set up.

**Review by antdevelopment1 — CHANGES_REQUESTED** [review]
Great job. Just let me know if it works on your end and if so no need to submit a new change.

File: scripts/index.js [hunk 2]
```diff
@@ -26,17 +26,65 @@ console.log (characters[0].died)
...
-        if (key.died.length > 0) {
-        deadList.push(characters.died);
-        }
-        
-    }
-    // console.log(key.died.length)
-    return deadList;
-})
-console.log(deadCharacters)
+function deadCharacters (characterList) {
```
Comment (antdevelopment1) [inline scripts/index.js:29]: Hey great job refactoring your deadCharcters function. Looks good.

File: scripts/index.js [hunk 3]
```diff
@@ -26,17 +26,65 @@ console.log (characters[0].died)
...
+        return personName.died !== '';
+    })
+    return deadArray.length;
+}
+console.log(deadCharacters(characters));
 
+// Who has the most titles?
+function mostTitles(characterList) {
+    characterList.sort(function(a,b) {
+        return (b.titles.length - a.titles.length)
```
Comment (antdevelopment1) [inline scripts/index.js:40]: Nice use of sort to target the person with the largest number of titles. Great job!!!

File: scripts/index.js [hunk 4]
```diff
@@ -26,17 +26,65 @@ console.log (characters[0].died)
...
+    })
+    return characterList[0].name;
+}
+console.log(mostTitles(characters));
 
+// How many are Valyrian?
+function areVarlyrian(characterList) {
+    let valyrianArray = characterList.filter(function(personName) {
```
Comment (antdevelopment1) [inline scripts/index.js:48]: Nice filter technique. Awesome stuff!!!

File: scripts/index.js [hunk 5]
```diff
@@ -26,17 +26,65 @@ console.log (characters[0].died)
...
+        return personName.culture === 'Valyrian';
+    })
+    return valyrianArray.length;
+}
+console.log(areVarlyrian(characters));
+
+//What actor plays "Hot Pie" (and don't use IMDB)?
+function isHotPie(characterList) {
```
Comment (antdevelopment1) [inline scripts/index.js:56]: Great job using filter to find Hot Pie.

File: scripts/index.js [hunk 6]
```diff
@@ -26,17 +26,65 @@ console.log (characters[0].died)
...
+    let hotPie = characterList.filter(function(personName) {
+        return personName.name === "Hot Pie";
+    })
+    return hotPie[0].playedBy;
+}
+console.log(isHotPie(characters));
+
+// How many characters are *not* in the tv show?
+function notInShow(characterList) {
+    let notInShowArray = characterList.filter(function(personName){
```
Comment (antdevelopment1) [inline scripts/index.js:66]: Great use of filter.

File: scripts/index.js [hunk 7]
```diff
@@ -26,17 +26,65 @@ console.log (characters[0].died)
...
+    })
+    return notInShowArray.length
+}
+console.log(notInShow(characters));
+
+
+// Produce a list characters with the last name "Targaryen"
+
+function isTargaryen(characterList) {
+    let targaryenArray = characterList.filter(function(personName){
```
Comment (antdevelopment1) [inline scripts/index.js:77]: Nice combo on using filter and map. Great work!!!

**Review by antdevelopment1 — COMMENTED** [review]
Great work overall. Really great layout for solutions.

File: scripts/index.js (hunk 1 above)
Comment (emauldin84) [reply scripts/index.js]: Hi April,

Yes, I am getting a return of 553 with both your solution and mine. See below for what I utilized. Thanks again for reviewing!

`function deadCharacters (characterList) {
    let deadArray = characterList.filter(function(personName) {
        return personName.died !== '';
    })
    return deadArray.length;
}
console.log(deadCharacters(characters));`

**Pull request closed without merging**